community/kernel/src/main/java/org/neo4j/kernel/api/txtracking/TransactionIdTracker.java

```diff
@@ -39,9 +40,11 @@
     private static final TimeUnit POLL_UNIT = TimeUnit.MILLISECONDS;
 
     private final TransactionIdStore transactionIdStore;
+    private final LongPredicate isAvailable;
 
-    public TransactionIdTracker( TransactionIdStore transactionIdStore )
+    public TransactionIdTracker( TransactionIdStore transactionIdStore, LongPredicate isAvailable )
```

Comment:
I would not call it a refactor. It is a one line change in `PlatformModule`. Just wrap creation of `AvailabilityGuard` with `dependencies.satisfyDependency(...)`.